firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/FirebaseGoogleAI.kt

```diff
@@ -0,0 +1,99 @@
+import com.google.firebase.vertexai.type.Tool
+import com.google.firebase.vertexai.type.ToolConfig
+
+public class FirebaseGoogleAI internal constructor(private val proxy: FirebaseVertexAI) {
+
+  /**
+   * Instantiates a new [GenerativeModel] given the provided parameters.
+   *
+   * @param modelName The name of the model to use, for example `"gemini-1.5-pro"`.
+   * @param generationConfig The configuration parameters to use for content generation.
+   * @param safetySettings The safety bounds the model will abide to during content generation.
+   * @param tools A list of [Tool]s the model may use to generate content.
+   * @param toolConfig The [ToolConfig] that defines how the model handles the tools provided.
+   * @param systemInstruction [Content] instructions that direct the model to behave a certain way.
+   * Currently only text content is supported.
+   * @param requestOptions Configuration options for sending requests to the backend.
+   * @return The initialized [GenerativeModel] instance.
+   */
+  @JvmOverloads
+  public fun generativeModel(
+    modelName: String,
+    generationConfig: GenerationConfig? = null,
+    safetySettings: List<SafetySetting>? = null,
+    tools: List<Tool>? = null,
+    toolConfig: ToolConfig? = null,
+    systemInstruction: Content? = null,
+    requestOptions: RequestOptions = RequestOptions(),
+  ): GenerativeModel =
+    proxy.generativeModel(
+      modelName,
+      generationConfig,
+      safetySettings,
+      tools,
+      toolConfig,
+      systemInstruction,
+      requestOptions,
+    )
+
+  /**
+   * Instantiates a new [ImagenModel] given the provided parameters.
+   *
+   * @param modelName The name of the model to use, for example `"imagen-3.0-generate-001"`.
+   * @param generationConfig The configuration parameters to use for image generation.
+   * @param safetySettings The safety bounds the model will abide by during image generation.
+   * @param requestOptions Configuration options for sending requests to the backend.
+   * @return The initialized [ImagenModel] instance.
+   */
+  @JvmOverloads
+  @PublicPreviewAPI
+  public fun imagenModel(
+    modelName: String,
+    generationConfig: ImagenGenerationConfig? = null,
+    safetySettings: ImagenSafetySettings? = null,
+    requestOptions: RequestOptions = RequestOptions(),
+  ): ImagenModel = proxy.imagenModel(modelName, generationConfig, safetySettings, requestOptions)
+
+  public companion object {
+    /** The [FirebaseGoogleAI] instance for the default [FirebaseApp] */
+    @JvmStatic
+    public val instance: FirebaseGoogleAI
+      get() = getInstance()
+
+    /** Returns the [FirebaseGoogleAI] instance for the provided [FirebaseApp]. */
+    @JvmStatic
+    public fun getInstance(app: FirebaseApp = Firebase.app): FirebaseGoogleAI {
+      val multiResourceComponent = app[FirebaseVertexAIMultiResourceComponent::class.java]
+      return FirebaseGoogleAI(multiResourceComponent.get(GenerativeBackend.GOOGLE_AI, "UNUSED"))
```

Comment:
In this case, we can have several, separate, instances of `FirebaseGoogleAI` which are backed by the same `proxy` class. For consistency, a 1:1 mapping would be better.